lib/models/device.coffee

```diff
@@ -275,23 +275,20 @@ exports.identify = (uuid, callback) ->
 # @param {String} newName - the device new name
 # @param {module:resin.models.device~renameCallback} callback - callback
 #
-# @todo This action doesn't return any error
-# if trying to rename a device that does not
-# exists. This should be fixed server side.
-#
 # @example
 #	resin.models.device.rename 317, 'NewName', (error) ->
 #		throw error if error?
 #		console.log("Device has been renamed!")
 ###
 exports.rename = (name, newName, callback) ->
-	return pine.patch
-		resource: 'device'
-		body:
-			name: newName
-		options:
-			filter:
-				name: name
+	exports.get(name).then ->
```

Comment:
Just saw this comment. I don't think SDK has to support something user-friendly. It's a programming interface, right? Nobody's using it by hand. If you search for your devices through the API you can get UUIDs as easy as the name. If you have the name in your app you can still do the same as the CLI — search the API through the SDK and pick the desired device then.